targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/TARGET_MCU_NRF52840/TARGET_EP_ATLAS/ONBOARD_TELIT_ME310.cpp

```diff
@@ -0,0 +1,216 @@
+}
+
+nsapi_error_t ONBOARD_TELIT_ME310::soft_power_off()
+{
+    onboard_modem_power_down();
+    return NSAPI_ERROR_OK;
+}
+
+nsapi_error_t ONBOARD_TELIT_ME310::init()
+{
+    nsapi_error_t err = AT_CellularDevice::init();
+    if (err != NSAPI_ERROR_OK) {
+        return err;
+    }
+    _at.lock();
+#if DEVICE_SERIAL_FC
+    _at.at_cmd_discard("&K3;&C1;&D0", "");
+#else
+    _at.at_cmd_discard("&K0;&C1;&D0", "");
+#endif
+
+    // AT#QSS=1
+    // Enable the Query SIM Status unsolicited indication in the ME. The format of the
+    // unsolicited indication is the following:
+    // #QSS: <status>
+    // The ME informs at
+    // every SIM status change through the basic unsolicited indication where <status> range is 0...1
+    // <status> values:
+    // - 0: SIM not inserted
+    // - 1: SIM inserted
+    _at.at_cmd_discard("#QSS", "=1");
+
+    // AT#PSNT=1
+    // Set command enables unsolicited result code for packet service network type (PSNT)
+    // having the following format:
+    // #PSNT:<nt>
+    // <nt> values:
+    // - 0: GPRS network
+    // - 4: LTE network
+    // - 5: unknown or not registered
+    _at.at_cmd_discard("#PSNT", "=1");
+
+    // AT+CMER=2
+    // Set command enables sending of unsolicited result codes from TA to TE in the case of
+    // indicator state changes.
+    // Current setting: buffer +CIEV Unsolicited Result Codes in the TA when TA-TE link is
+    // reserved (e.g. on-line data mode) and flush them to the TE after
+    // reservation; otherwise forward them directly to the TE
+    _at.at_cmd_discard("+CMER", "=2");
+
+    // AT+CMEE=2
+    // Set command disables the use of result code +CME ERROR: <err> as an indication of an
+    // error relating to the +Cxxx command issued. When enabled, device related errors cause the +CME
+    // ERROR: <err> final result code instead of the default ERROR final result code. ERROR is returned
+    // normally when the error message is related to syntax, invalid parameters or DTE functionality.
+    // Current setting: enable and use verbose <err> values
+    _at.at_cmd_discard("+CMEE", "=2");
+
+    // AT#PORTCFG=0
+    // Set command allows to connect Service Access Points to the external physical ports giving a great
+    // flexibility. Examples of Service Access Points: AT Parser Instance #1, #2, #3, etc..
+    _at.at_cmd_discard("#PORTCFG", "=", "%d", EP_ATLAS_PORT_CONFIGURATION_VARIANT);
+
+    // AT&W&P
+    // - AT&W: Execution command stores on profile <n> the complete configuration of the device. If
+    //         parameter is omitted, the command has the same behavior of AT&W0.
+    // - AT&P: Execution command defines which full profile will be loaded at startup. If parameter
+    //         is omitted, the command has the same behavior as AT&P0
+    _at.at_cmd_discard("&W&P", "");
+
+    return _at.unlock_return_error();
+}
+
+void ONBOARD_TELIT_ME310::press_power_button(int time_ms)
+{
+    gpio_t gpio_CELL_ON_OFF;
+    //gpio_t gpio_PWR_MON;
```

Comment:
is this dead code that should not be here? There's lot of code without any further comment.

Reply:
Dead code removed.  Thanks!